heuristic/alns.py

```diff
@@ -158,35 +169,71 @@ def __init__(self, state, data, criterion):
+
+        # repair_worst_employee_greedy = partial(worst_employee_repair, self.competencies, self.t_covered_by_shift, 
+        #                                         self.employee_with_competencies, self.employee_with_competency_combination,
+        #                                         self.demand, self.contracted_hours, self.weeks, self.time_periods_in_week,
+        #                                         self.time_step, self.shifts, self.shifts_at_day)
+
+        #self.add_destroy_operator([remove_worst_employee, remove_worst_week])
+        #self.add_repair_operator([repair_worst_week_regret, repair_worst_employee_regret, repair_worst_week_greedy, repair_worst_employee_greedy])
+        #
+        #repair_worst_week_regret, 
         operators = {
-            remove_worst_employee: [repair_worst_employee_regret, repair_worst_employee_greedy],
-            remove_worst_week: [repair_worst_week_regret, repair_worst_week_greedy],
-        }
+                        remove_worst_employee: [repair_worst_employee_regret, repair_worst_employee_greedy],
+                        remove_worst_week: [repair_worst_week_regret, repair_worst_week_greedy]
+                    }
         self.add_destroy_and_repair_operators(operators)
+        #for key in self.repair_operators.keys():
+         #   for operator in self.repair_operators[key]:
+         #       print(str(key) + ": " + str(operator))
         self.initialize_destroy_and_repair_weights()
+        #print(self.destroy_weights)
+        #print(self.repair_weights)
+
 
     def iterate(self, iterations):
         for iteration in range(iterations):
             candidate_solution = self.current_solution.copy()
 
-            destroy_operator, destroy_operator_id = self.select_operator(
-                self.destroy_operators, self.destroy_weights
-            )
-            repair_operator, repair_operator_id = self.select_operator(
-                self.repair_operators[destroy_operator_id], self.repair_weights[destroy_operator_id]
-            )
+            destroy_operator, destroy_operator_id = self.select_operator(self.destroy_operators, self.destroy_weights)
+            repair_operator, repair_operator_id = self.select_operator(self.repair_operators[destroy_operator_id], self.repair_weights[destroy_operator_id])
 
             destroy_set, destroy_specific_set = destroy_operator(candidate_solution)
             repair_set = repair_operator(candidate_solution, destroy_set, destroy_specific_set)
 
             self.calculate_objective(candidate_solution, destroy_set, repair_set)
-            self.consider_candidate_and_update_weights(
-                candidate_solution, destroy_operator_id, repair_operator_id
-            )
+            self.consider_candidate_and_update_weights(candidate_solution, destroy_operator_id, repair_operator_id)
 
-            candidate_solution = self.current_solution.copy()
```

Comment:
Is this on purpose?

Reply:
Yes it is on purpose. I believe through another merge this has been introduced by mistake. Why would you have another copy of the state at the bottom when it the next line in the loop says the exact same thing?